server/src/main/java/org/elasticsearch/common/xcontent/XContent.java

```diff
@@ -52,7 +51,15 @@
      */
     static boolean isStrictDuplicateDetectionEnabled() {
         // Don't allow duplicate keys in JSON content by default but let the user opt out
-        return Booleans.parseBoolean(System.getProperty("es.xcontent.strict_duplicate_detection", "true"));
+        String dupProp = System.getProperty("es.xcontent.strict_duplicate_detection", "true");
```

Comment:
Can you make this one separate?

Reply:
A separate PR?

Reply:
Yes please, the change has nothing to do with the title (it does with the goal, but not the title).

Reply:
Certainly, thanks for clarifying